src/compiler/resolutionCache.ts

```diff
@@ -713,6 +714,10 @@ namespace ts {
                     if (!isPathWithDefaultFailedLookupExtension(fileOrDirectoryPath) && !customFailedLookupPaths.has(fileOrDirectoryPath)) {
                         return false;
                     }
+                    // prevent saving an open file from over-eagerly triggering invalidation

```

Comment:
Also I think you should move this right after `if (isPathIgnored(fileOrDirectoryPath)) return false;` so that it is skipped in all cases

Reply:
Moved.